### ISSUE TITLE
Issue with Performance Drop After Fine-Tuning

### ISSUE THREAD

**Issue by hkgdifyu** [seal-rg/recurrent-pretraining#22]
Hello, your work has been very inspiring to me. I tried fine-tuning your model using the `llamafactory` library, with the following key parameters:，However, the current results show a performance drop on both arc_challenge and arc_easy. Specifically:

1. Before fine-tuning (at num_steps=40): Scores were 34 and 68.
2. After fine-tuning: Scores dropped to 20 and 30 (worse than even the fine-tuned num_steps=8 results of 21 and 39).

Have you encountered this issue after fine-tuning, or could there be a problem with my configuration? 

```yaml
# Training Config
stage: sft  
do_train: true  
finetuning_type: lora  
lora_rank: 8  
lora_target: all  

per_device_train_batch_size: 2  
gradient_accumulation_steps: 32  
learning_rate: 1.0e-5  
num_train_epochs: 3.0  
lr_scheduler_type: cosine  
warmup_ratio: 0.1  
bf16: false  
ddp_timeout: 180000000  
resume_from_checkpoint: null  
disable_gradient_checkpointing: true  
flash_attn: disabled  

# Dataset Config
dataset: openorca, mathinstruct, alpaca_gpt4_en, codealpaca  
template: default  
cutoff_len: 2048  
max_samples: 1000  
overwrite_cache: true  
preprocessing_num_workers: 16  
dataloader_num_workers: 4

**Comment by JonasGeiping** [comment]
Hi!

In general, finetuning this model is still a bit under-developed and not as straightforward as for standard models. Regarding your config, it looks sensible to me, but are the loss curves looking ok? 

Also, where is the Lora being injected? I'm surprised llama factory can handle the non-standard architecture of this model.
